src/constructor/constructor.py

```diff
@@ -0,0 +1,259 @@
+from collections.abc import Iterable
+from typing import Any, Dict, List, Optional, Union     # noqa: H301
+
+import albumentations as A
+from omegaconf import DictConfig, ListConfig
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from src.constructor import DATASETS, LOSSES, OPTIMIZERS, SCHEDULERS, TRANSFORMS
+from src.data.datasets.base import ImageDataset
+from src.losses.base import JointLoss
+
+
+class Constructor:
+    """Provides factory features for optimizers, schedulers, loss functions, data loaders and metrics."""
+
+    def __init__(self, hparams: DictConfig):
+        """Init Constructor with hparams.
+
+        Args:
+            hparams: Configuration dictionary for all the mentioned components.
+            The dictionary should contain at least the following parameter groups (see configuration for details):
+            - optimization
+            - data
+            - losses
+            - metrics
+        """
+        self.__hparams = hparams
+
+    def configure_optimizers(self, parameters: Union[Module, Tensor, Iterable[Union[Module, Tensor]]],
```

Comment:
Actually don't clearly understand how it will be called. Do you want to set all Task - LightningModule as input for configure optimizers, or you want to set constructor as parameter of Task? For me both scenarios is bad...